### PR TITLE
Vt checkins

### PR DESCRIPTION
**Pull request by github-gcontributor**
Hi @moidx 
sw/device/tests/aes_basic_enc_dec.c
Can you let me know your inputs on the 
1. code structure
2. Commands used are in line with code
3. Sequence of commands to perform encryption
4. Sequence of command to perform decryption
5. Comparison mechanism is correct

https://github.com/lowRISC/opentitan/commit/6357d927ef8fbde9f4841ba52fb0693f1e91ccf3
For the Initial checken for the test aes_force_prng_reseed, stall signal is not getting asserted.
Try1: Tried with basic enc dec sequence of steps and disabling edn uisng aes_edn_stop() with manual mode as well as automatic mode
Try2: Leverage part of code form aes_serial.c, and then followed to disable edn.
Please suggest if I missing any other steps or sequence

### CODE REVIEW

**Comment by github-gcontributor** [comment]
https://github.com/github-gcontributor/opentitan/blob/vt_checkins/sw/device/tests/aes_force_prng_reseed_test.c

@moidx Updated code in line with description.
(ignoring/skipping the stall bit getting asserted) and proceeding further... 
using macros to test optionally enable/disable
RESEED_TEST_EN 1 
DEBUG_EN 0 
TRYSETTING1_EN 1  --> this code used to reconfigure the edn/csrng/entropy after aes_start and before prng_reseed.

File: rules/opentitan/defs.bzl [hunk 1]
```diff
@@ -108,6 +108,16 @@ EARLGREY_CW340_TEST_ENVS = {
...
+EARLGREY_SIM_VERILATOR_ENVS = {
+    "//hw/top_earlgrey:sim_verilator": None,
+}
+################################varunkumart@###################################
+EARLGREY_CW340_CUSTOM_TEST_ENVS = {
+    "//hw/top_earlgrey:fpga_cw340_sival": None,  # Updated for CW340
+    "//hw/top_earlgrey:fpga_cw340_rom_with_fake_keys": None,  # Updated for CW340
+}
+################################varunkumart@###################################
 # Messages we expect for possible test outcomes.
```
Comment (moidx) [inline rules/opentitan/defs.bzl:121]: Please revert this change.

File: sw/device/tests/BUILD [hunk 2]
```diff
@@ -38,6 +38,27 @@ load(
...
+    "//rules/opentitan:defs.bzl",
+    "EARLGREY_SIM_VERILATOR_ENVS",
+)
+################################varunkumart@###################################
+load(
+    "//rules/opentitan:defs.bzl",
+    "EARLGREY_CW340_CUSTOM_TEST_ENVS",
+)
+################################varunkumart@###################################
+
```
Comment (moidx) [inline sw/device/tests/BUILD:60]: Please revert this change.

File: sw/device/tests/BUILD [hunk 3]
```diff
@@ -153,6 +174,93 @@ opentitan_test(
...
+## cc_library(
+##     name = "aes_example_lib",
+##     srcs = ["hw/ip/aes/model/aes_example.c"],
+##     hdrs = ["hw/ip/aes/model/aes_example.h"],
+##     deps = [
+##         "//sw/device/lib/base:base",
+##         "//sw/device/lib/dif:dif_aes",
+##         "//sw/device/lib/runtime:log",
+##     ],
+## )
```
Comment (moidx) [inline sw/device/tests/BUILD:188]: Please remove dead code.

File: sw/device/tests/BUILD [hunk 4]
```diff
@@ -153,6 +174,93 @@ opentitan_test(
...
+    srcs = ["aes_force_prng_reseed_test.c"],
+    verilator = verilator_params(
+        timeout = "long",
+    ),
+    exec_env = dicts.add(
+        {
+            "//hw/top_earlgrey:sim_verilator": None,
+            ##"//hw/top_earlgrey:fpga_cw340_sival": None,
+            ##"//hw/top_earlgrey:fpga_cw340_sival_rom_ext": None,
+            ## "//hw/top_earlgrey:silicon_creator": None,
```
Comment (moidx) [inline sw/device/tests/BUILD:202]: Please remove dead code.

File: sw/device/tests/aes_force_prng_reseed_test.c [hunk 5]
```diff
@@ -0,0 +1,509 @@
...
+  LOG_INFO("REGISTER AES_CTRL_AUX_REGWEN: 0x%x", ctrl_aux_regwen);
+}
+
+
+enum {
+  kTestTimeout = 1000 * 1000,  // Timeout for waiting for AES operations
+};
+
+static  dif_aes_t aes;
+#if RESEED_TEST_EN
```
Comment (moidx) [inline sw/device/tests/aes_force_prng_reseed_test.c:77]: Please remove #defines. These are not allowed by the style guide. Please also remove all other #defines in this PR.

https://opentitan.org/book/doc/contributing/style_guides/c_cpp_coding_style.html

File: sw/device/tests/aes_force_prng_reseed_test.c [hunk 6]
```diff
@@ -0,0 +1,509 @@
+// Copyright lowRISC contributors (OpenTitan project).
```
Comment (moidx) [inline sw/device/tests/aes_force_prng_reseed_test.c:1]: Please remove all dead code, and run the following command to format the file: 

```
bazel run //quality:clang_format_fix
```

File: sw/device/tests/aes_force_prng_reseed_test.c [hunk 7]
```diff
@@ -0,0 +1,509 @@
...
+                                       // software manually triggers the
+                                       // operation
+   .reseed_on_key_change = true,  // Enable PRNG reseeding every time the key
+                                  // is changed (KEY_TOUCH_FORCES_RESEED)
+   // Setting force_masks to false as per disucsseion
+   .force_masks = false,     // Enable forced masking for the key (important for
+                            // security during testing)
+   .ctrl_aux_lock = false,  // Do not lock the auxiliary control settings,
+                             // allowing further modifications
+  };
```
Comment (moidx) [inline sw/device/tests/aes_force_prng_reseed_test.c:244]: Assembly style comments are also not aligned with the style guide. Many of the comments here can be removed as they don't add any explanation over the enum names.

File: sw/device/tests/aes_force_prng_reseed_test.c [hunk 8]
```diff
@@ -0,0 +1,509 @@
...
+
+  //
+  return OK_STATUS();
+}
+
+bool test_main(void) {
+  return status_ok(execute_test()); // Execute the test and return status
+}
+
+/*
```
Comment (moidx) [inline sw/device/tests/aes_force_prng_reseed_test.c:442]: Please remove log output.

File: sw/device/tests/BUILD [hunk 9]
```diff
@@ -2457,6 +2711,24 @@ opentitan_test(
...
+    deps = [
+        "//hw/top_earlgrey/sw/autogen:top_earlgrey",
+        "//sw/device/lib/base:csr",
+        "//sw/device/lib/runtime:hart",
+        "//sw/device/lib/runtime:irq",
+        "//sw/device/lib/runtime:pmp",
+        "//sw/device/lib/testing/test_framework:ottf_main",
+    ],
+)
+################################varunkumart@###################################
```
Comment (moidx) [inline sw/device/tests/BUILD:2730]: Please remove.

File: sw/device/tests/BUILD [hunk 10]
```diff
@@ -2416,6 +2524,152 @@ opentitan_test(
...
+## nw ##         "//sw/device/lib/base:macros",
+## nw ##         "//sw/device/lib/base:memory",
+## nw ##         "//sw/device/lib/dif:otp_ctrl",
+## nw ##         "//sw/device/lib/runtime:log",
+## nw ##         "//sw/device/lib/testing:otp_ctrl_testutils",
+## nw ##         "//sw/device/lib/testing/test_framework:ottf_main",
+## nw ##     ],
+## nw ## )
+
+################################varunkumart@###################################
```
Comment (moidx) [inline sw/device/tests/BUILD:2587]: Please remove all commented out targets.

File: sw/device/tests/BUILD [hunk 11]
```diff
@@ -2416,6 +2524,152 @@ opentitan_test(
...
+        "//sw/device/lib/base:bitfield",
+        "//sw/device/lib/base:macros",
+        "//sw/device/lib/base:memory",
+        "//sw/device/lib/dif:otp_ctrl",
+        "//sw/device/lib/runtime:log",
+        "//sw/device/lib/testing:otp_ctrl_testutils",
+        "//sw/device/lib/testing/test_framework:ottf_main",
+    ],
+)
+################################varunkumart@###################################
```
Comment (moidx) [inline sw/device/tests/BUILD:2545]: Please avoid using these guard comments.

File: sw/device/tests/BUILD
Comment (moidx) [inline sw/device/tests/BUILD:1]: These file contains many changes that are unnecessary. I expect to see exactly two new targets and nothing else. Thanks

File: sw/device/tests/aes_basic_enc_dec_manualmode.c [hunk 12]
```diff
@@ -0,0 +1,226 @@
...
+OTTF_DEFINE_TEST_CONFIG();
+
+// Register Offsets based on the OpenTitan documentation
+#define AES_CTRL_SHADOWED_REG_OFFSET         0x74
+#define AES_CTRL_AUX_SHADOWED_REG_OFFSET     0x78
+#define AES_CTRL_AUX_REGWEN_REG_OFFSET       0x7C
+#define AES_TRIGGER_REG_OFFSET               0x80
+#define AES_STATUS_REG_OFFSET                0x84
+#define AES_CTRL_AUX_SHADOWED_KEY_TOUCH_FORCES_RESEED_BIT  0
+#define AES_CTRL_AUX_SHADOWED_FORCE_MASKS_BIT              1
```
Comment (moidx) [inline sw/device/tests/aes_basic_enc_dec_manualmode.c:26]: Please remove these registers definitions. All access to registers must be done in dif libraries. See dif_aes.c for more details.

File: sw/device/tests/aes_basic_enc_dec_manualmode.c [hunk 13]
```diff
@@ -0,0 +1,226 @@
...
+  uint32_t ctrl_aux_regwen = mmio_region_read32(aes_base, AES_CTRL_AUX_REGWEN_REG_OFFSET);
+  uint32_t trigger = mmio_region_read32(aes_base, AES_TRIGGER_REG_OFFSET);
+  uint32_t status = mmio_region_read32(aes_base, AES_STATUS_REG_OFFSET);
+
+  LOG_INFO("REGISTER AES_CTRL_SHADOWED: 0x%x", ctrl_shadowed);
+  LOG_INFO("REGISTER AES_CTRL_AUX_SHADOWED: 0x%x", ctrl_aux_shadowed);
+  LOG_INFO("REGISTER AES_STATUS: 0x%x", status);
+  LOG_INFO("REGISTER AES_TRIGGER: 0x%x", trigger);
+  LOG_INFO("REGISTER AES_CTRL_AUX_REGWEN: 0x%x", ctrl_aux_regwen);
+}
```
Comment (moidx) [inline sw/device/tests/aes_basic_enc_dec_manualmode.c:40]: Please remove this function as well. Explore using difs for this purpose.

File: sw/device/tests/aes_basic_enc_dec_manualmode.c [hunk 14]
```diff
@@ -0,0 +1,226 @@
...
+
+  // Monitor AES registers
+  LOG_INFO("AES registers after hardware initialized and reset:");
+  monitor_aes_registers(mmio_region_from_addr(TOP_EARLGREY_AES_BASE_ADDR));
+
+  // Configure AES for ECB mode, 128-bit key, and manual operation
+  // Define the transaction settings for the AES module
+  LOG_INFO("AES transaction to be configured as: Operation=Encrypt, Mode=ECB, Key Length=128-bit, Manual Operation, reseed=true and force_masks=True for our desired operation");
+
+dif_aes_transaction_t transaction = {
```
Comment (moidx) [inline sw/device/tests/aes_basic_enc_dec_manualmode.c:90]: assembly style comments are not allowed.

File: sw/device/tests/aes_basic_enc_dec_manualmode.c [hunk 15]
```diff
@@ -0,0 +1,226 @@
...
+
+
+
+  // "Convert" key share byte arrays to `dif_aes_key_share_t`
+  // Setup the key with one share being `kAesModesKey128` and the other share as zero
+  dif_aes_key_share_t key;
+  memcpy(key.share0, kAesModesKey128, sizeof(key.share0)); // Share 0 is the test key
+  memset(key.share1, 0, sizeof(key.share1));        // Share 1 is zero
+
+  // Debug: Print key being used for the encryption
```
Comment (moidx) [inline sw/device/tests/aes_basic_enc_dec_manualmode.c:123]: Please remove comments that are redundant. Log messages are self-explanatory.

**Review by moidx — COMMENTED** [review]
Thanks for submitting this PR. Before I can do I full review, please 

1. Address all the style guide and high level comments. i.e. Removed unnecessary files from this PR. Remove dead-code, etc.
2. Run C/C++ formatting: `bazel run //quality:clang_format_fix`
3. Ammed your commit and force push:

```console
# Stage files
git add .
git commit --amend -s
git push -f origin a1-sival-aes-tests
```

**Comment by moidx** [comment]
Please also squash all commits into one, and add a detail description of the changes to the commit message, and update the PR title and description as well. 

Thanks
